requests/utils.py

```diff
@@ -732,6 +732,22 @@ def to_native_string(string, encoding='ascii'):
 
     return out
 
+def validate_header(header):
+    """Verifies that header value doesn't contain leading whitespace or
+    return characters. This prevents unintended header injection.
+
+    header is a tuple, in the format (name, value)
+    """
+    name, value = header
+
+    if isinstance(value, bytes):
+        pat = re.compile(b'^\S[^\\r\\n]*$|^$')
+    else:
+        pat = re.compile(r'^\S[^\r\n]*$|^$')
```

Comment:
So using `re.compile` is great here, but we lose the advantage because these are done in the function. Can you hoist both these REs up to module scope and give them neat names, and then have this function choose between them?